src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java

```diff
@@ -232,27 +227,74 @@ public Claim saveClaim(
             .build();
 
         Claim savedClaim = caseRepository.saveClaim(user, claim);
+        String pin = pinResponse.map(GeneratePinResponse::getPin).orElse(null);
+        createClaimEvent(authorisation, user, pin, savedClaim);
+        trackClaimIssued(savedClaim.getReferenceNumber(), savedClaim.getClaimData().isClaimantRepresented());
+
+        return savedClaim;
+    }
+
+    @LogExecutionTime
+    @Transactional(transactionManager = "transactionManager")
+    public Claim saveLegalRepClaim(
+        String submitterId,
+        ClaimData claimData,
+        String authorisation
+    ) {
+        String externalId = claimData.getExternalId().toString();
+        User user = isExistingClaim(authorisation, externalId);
+
+        Optional<GeneratePinResponse> pinResponse = getPinResponse(claimData, authorisation);
+        Optional<String> letterHolderId = pinResponse.map(GeneratePinResponse::getUserId);
```

Comment:
I'd put the `.orElse(null)` here so that it's clear what this is for, rather than a few lines down when building the claim

Reply:
I agree, I have done as you said but reverted back to match what is done for saveClaim method but i think i should probably correct in both places. I will do that now.